src/main/java/com/hibernate/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,70 @@
+package com.hibernate.cinema.dao.impl;
+
+import com.hibernate.cinema.dao.ShoppingCartDao;
+import com.hibernate.cinema.exception.DataProcessingException;
+import com.hibernate.cinema.lib.Dao;
+import com.hibernate.cinema.model.ShoppingCart;
+import com.hibernate.cinema.model.User;
+import com.hibernate.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add shopping cart "
+                    + shoppingCart + "to database :", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from ShoppingCart sc left join"
+                    + " fetch sc.tickets where sc.user = :user", ShoppingCart.class)
```

Comment:
```suggestion
            return session.createQuery("from ShoppingCart sc "
                    + "left join fetch sc.tickets where sc.user = :user", ShoppingCart.class)
```
You could also add `join fetch sc.user` to prevent the creation of an extra query for users